pkg/kubelet/kubelet.go

```diff
@@ -699,11 +699,16 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		}
 		// TODO(roberthbailey): Can we do this without having credentials to talk
 		// to the cloud provider?
-		instanceID, err := instances.ExternalID(kl.hostname)
+		externalID, err := instances.ExternalID(kl.hostname)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get external ID from cloud provider: %v", err)
+		}
+		node.Spec.ExternalID = externalID
+		instanceID, err := instances.InstanceID(node.Name)
```

Comment:
One problem with this (which is pre-existing) is that we can't call into the public cloudprovider API from the node. We should only call local metadata APIs, at most. getInstanceID and getProjectAndZone work this way for GCE, but it doesn't look like the AWS implementation does.

Please strengthen the TODO above:
```
// TODO: We can't assume that the node has credentials to talk to the cloudprovider from arbitrary nodes.
// At most, we should talk to a local metadata server here.
```